wandb/sdk/launch/agent/agent.py

```diff
@@ -331,6 +332,11 @@ def finish_thread_id(
                 ):
                     break
                 if run_info is None:
+                    # Fetch the logs now if we don't get run info on the
+                    # first try, in case the logs are cleaned from the runner
+                    # environment (e.g. k8s) during the run info grace period.
+                    if interval == 1:
```

Comment:
There's no way the logs would only be available later right? If so, we could say `if logs is None`. Otherwise this is fine